compiler/native/environment.go

```diff
@@ -73,15 +80,22 @@ func (c *client) EnvironmentSteps(s yaml.StepSlice) (yaml.StepSlice, error) {
 func (c *client) EnvironmentServices(s yaml.ServiceSlice) (yaml.ServiceSlice, error) {
 	// iterate through all services
 	for _, service := range s {
+		env := make(map[string]string)
 		// gather set of default environment variables
-		env := environment(c.build, c.metadata, c.repo, c.user)
+		defaultEnv := environment(c.build, c.metadata, c.repo, c.user)
 
 		// inject the declared environment
 		// variables to the build service
 		for k, v := range service.Environment {
 			env[k] = v
 		}
 
+		// inject the default environment
+		// variables to the build service
```

Comment:
same as other comment

Reply:
Added comments to hopefully clarify this.